dbms/src/Functions/FunctionsString.cpp

```diff
@@ -4381,6 +4381,420 @@ class FunctionRepeat : public IFunction
     }
 };
 
+class FunctionInstrUTF8 : public IFunction
+{
+public:
+    static constexpr auto name = "instrUTF8";
+    static constexpr auto diff = 'Z' - 'z';
+
+    FunctionInstrUTF8() = default;
+
+    static FunctionPtr create(const Context & /*context*/)
+    {
+        return std::make_shared<FunctionInstrUTF8>();
+    }
+
+    std::string getName() const override { return name; }
+    size_t getNumberOfArguments() const override { return 2; }
+
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        auto first_argument = removeNullable(arguments[0]);
+        if (!first_argument->isString())
+            throw Exception(
+                fmt::format("Illegal type {} of first argument of function {}", arguments[0]->getName(), getName()),
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        auto second_argument = removeNullable(arguments[1]);
+        if (!second_argument->isString())
+            throw Exception(
+                fmt::format("Illegal type {} of first argument of function {}", arguments[1]->getName(), getName()),
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+        auto return_type = std::make_shared<DataTypeInt64>();
+        return (arguments[0]->isNullable() || arguments[1]->isNullable()) ? makeNullable(return_type) : return_type;
+    }
```

Comment:
for `useDefaultImplementationForNulls() const { return true; }`, we don't need to consider nullable
```suggestion
    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
    {
        auto first_argument = arguments[0];
        if (!first_argument->isString())
            throw Exception(
                fmt::format("Illegal type {} of first argument of function {}", arguments[0]->getName(), getName()),
                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
        auto second_argument = arguments[1];
        if (!second_argument->isString())
            throw Exception(
                fmt::format("Illegal type {} of first argument of function {}", arguments[1]->getName(), getName()),
                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);

        return std::make_shared<DataTypeInt64>();
    }
```